Streamlining API Authentication (#251)
NVIDIA/NeMo-Agent-Toolkit: docs/source/extend/adding-an-authentication-provider.md

<!--
SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
SPDX-License-Identifier: Apache-2.0

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->

> # Adding an API Authentication Provider to NVIDIA Agent Intelligence Toolkit

Reviewer: ```suggestion
# Adding an API Authentication Provider to NVIDIA NeMo Agent Toolkit
```

Reviewer: Please change all mentions of Agent Intelligence or AIQ toolkit to NeMo Agent toolkit.

:::{note}
We recommend reading the [Streamlining API Authentication](../reference/api-authentication.md) guide before proceeding with
this detailed documentation.
:::

> The AIQ Toolkit offers a set of built-in authentication providers for accessing API resources. Additionally, it includes

Reviewer: ```suggestion
The NeMo Agent toolkit offers a set of built-in authentication providers for accessing API resources. Additionally, it includes
```

Reviewer: "t" in toolkit should be lower case

                                         client_secret="test_secret_secure_16_chars_minimum",
                                         client_id="test_client",
                                         audience="test_audience",
                                         scope=["test_scope"])

    auth_client = AuthCodeGrantClient(config=minimal_config, config_name="test_config")

    tester = OAuth2FlowTester(oauth_client=auth_client, flow=OAuth2Flow.AUTHORIZATION_CODE)

    # Run the complete OAuth2 flow test suite
    assert await tester.run() is True
```

## Packaging the Provider and Client

The provider and client will need to be bundled into a Python package, which in turn will be registered with AIQ
toolkit as a [plugin](../extend/plugins.md). In the `pyproject.toml` file of the package the
`project.entry-points.'aiq.components'` section, defines a Python module as the entry point of the plugin. Details on
how this is defined are found in the [Entry Point](../extend/plugins.md#entry-point) section of the plugins document.
By convention, the entry point module is named `register.py`, but this is not a requirement.

> In the entry point module it is important that the provider is defined first followed by the client, this ensures that

Reviewer: ```suggestion
In the entry point module, it is important that the provider is defined first followed by the client. This ensures that
```
